core/src/main/java/com/linecorp/armeria/server/ArmeriaDefaultHttpRequest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Map;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.MoreObjects.ToStringHelper;
+
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.RequestHeadersBuilder;
+
+import io.netty.handler.codec.http.DefaultHttpMessage;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.util.internal.ObjectUtil;
+
+final class ArmeriaDefaultHttpRequest extends DefaultHttpMessage implements HttpRequest {
+    private static final int HASH_CODE_PRIME = 31;
+    private HttpMethod method;
+    private final RequestHeadersBuilder builder;
+    private final HttpHeaders headers;
+    private String uri;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param httpVersion       the HTTP version of the request
+     * @param method            the HTTP method of the request
+     * @param uri               the URI or path of the request
+     * @param validateHeaders   validate the header names and values when adding them to the {@link HttpHeaders}
+     */
+    ArmeriaDefaultHttpRequest(HttpVersion httpVersion, HttpMethod method, String uri,
+                                     boolean validateHeaders) {
+        super(httpVersion, validateHeaders, false);
+        this.method = checkNotNull(method, "method");
+        this.uri = checkNotNull(uri, "uri");
+        builder = RequestHeaders.builder(
+                com.linecorp.armeria.common.HttpMethod.valueOf(method.name()), uri);
```

Comment:
> But I couldn't figure out 1)why the request is mapped with DefaultFullHttpRequest and 2)the location where it's mapped.
@minwoox Can you provide some advice regarding it?

It seems like an exception is raised while converting the method. We need to fix it:
```suggestion
        final com.linecorp.armeria.common.HttpMethod armeriaMethod =
                com.linecorp.armeria.common.HttpMethod.tryParse(method.name());
        builder = RequestHeaders.builder(armeriaMethod != null ? armeriaMethod
                                                               : com.linecorp.armeria.common.HttpMethod.UNKNOWN,
                                         uri);
```